ClientFactory/ClientFactoryInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Client\HttpClient;
+
+/**
+ * Interface ClientFactoryInterface.
+ */
+interface ClientFactoryInterface
+{
+    /**
+     * Configure and return a client.
+     *
+     * @param array $config
+     *
+     * @return HttpClient
```

Comment:
My intention was to include the scenario where a clients configuration was wrong and we could not create a client because of it.